hbase-client/src/main/java/org/apache/hadoop/hbase/client/CatalogReplicaLoadBalanceSimpleSelector.java

```diff
@@ -262,16 +261,16 @@ private void cleanupReplicaReplicaStaleCache() {
   private int refreshCatalogReplicaCount() {
     int newNumOfReplicas = this.getNumOfReplicas.getAsInt();
     LOG.debug("Refreshed replica count {}", newNumOfReplicas);
-    if (newNumOfReplicas == 1) {
-      LOG.warn("Table {}'s region replica count is 1, maybe a misconfiguration or failure to "
-        + "fetch the replica count", tableName);
+    // If the returned number of replicas is -1, it is caused by failure to fetch the
+    // replica count. Do not update the numOfReplicas in this case.
+    if (newNumOfReplicas == CatalogReplicaLoadBalanceSelector.UNINITIALIZED_NUM_OF_REPLICAS) {
+      LOG.error("Failed to fetch Table {}'s region replica count", tableName);
```

Comment:
How often will we log this message when a client is trying to connect to the cluster? Will it happen frequently?

Reply:
Currently, it is logged every 2 minutes.